src/imars3d/backend/workflow/engine.py

```diff
@@ -43,7 +45,7 @@ def validate_type(outputs):
                 raise WorkflowEngineError("Task and Function outputs cannot be string")
 
         validate_type(task_outputs)
-        if function_outputs:
+        if function_outputs is not None:
```

Comment:
Good catch! the previous `if` will return `False` for empty list but empty lists should be checked